### PR TITLE
Replace sha1 and changelist just if it's required

### PR DESCRIPTION
**Pull request by ayoukhananov**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,6 +72,7 @@
     <maven-source-plugin.version.a>2.2.1</maven-source-plugin.version.a>
     <maven-compiler-plugin.version>3.5.1</maven-compiler-plugin.version>
     <maven-scm.version>1.11.2</maven-scm.version>
+    <junit.version>4.13</junit.version>
```
Comment (amirhadadi) [inline pom.xml:75]: why not junit 5?

File: src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/FlattenMojo.java [hunk 2]
```diff
@@ -43,15 +45,17 @@
...
    * {@inheritDoc}
    */
   public void execute() throws MojoExecutionException {
     final String originalPom = readPom();
     final String revision = getRevision();
-    final String modifiedPom = replacePlaceHolders(originalPom, revision, sha1, changeList);
+    final String modifiedPom = pomVisitor.visit(originalPom, revision, sha1, changeList);
     if (originalPom.equals(modifiedPom)) {
-      getLog().info("POM is not CI friendly");
+      getLog().info("POM is not required CI friendly flatten operation");
```
Comment (amirhadadi) [inline src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/FlattenMojo.java:58]: Pom does not have any CI friendly properties

File: src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitor.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.outbrain.ci.friendly.flatten.maven.plugin.visitor;
+
+public interface PomVisitor {
```
Comment (amirhadadi) [inline src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitor.java:3]: Why do you need an interface? there's only one implementation.

File: src/test/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitorImplTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import org.apache.commons.io.FileUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+public class PomVisitorImplTest {
```
Comment (amirhadadi) [inline src/test/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitorImplTest.java:12]: Liked the test

File: pom.xml (hunk 1 above)
Comment (ayoukhananov) [reply pom.xml:75]: Will do

File: src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/FlattenMojo.java (hunk 2 above)
Comment (ayoukhananov) [reply src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/FlattenMojo.java:58]: will change

File: src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitor.java (hunk 3 above)
Comment (ayoukhananov) [reply src/main/java/com/outbrain/ci/friendly/flatten/maven/plugin/visitor/PomVisitor.java:3]: In the beginning i thought about visitor for each property . But then i aggregate them into one .
Might be will split them again in the near future

**Pull request merged**